autocorpus/table.py

```diff
@@ -6,565 +6,583 @@
+    if not temp_row:
+        return None
+    temp_cells: list[Tag] = (
+        temp_row.findAll(["th", "td"]) if isinstance(temp_row, Tag) else []
+    )
+    n_cols: int = (
+        sum([int(i.attrs["colspan"]) for i in temp_cells]) if temp_cells else 0
+    )
+
+    # build an empty matrix for all possible cells
+    table: list[list[str]] = [[""] * n_cols for row in rows]
+
+    # fill matrix from row data
+    rowspans: dict[
+        int, int
+    ] = {}  # track pending rowspans, column number mapping to count
+    for row_idx, row in enumerate(rows):
+        span_offset: int = 0  # how many columns are skipped due to row and colspans
+        for col_idx, cell in enumerate(row.findAll(["td", "th"])):
+            # adjust for preceding row and colspans
+            col_idx += span_offset
+            while rowspans.get(col_idx, 0):
+                span_offset += 1
+                col_idx += 1
+
+            # fill table data
+            rowspan = int(cell.attrs["rowspan"])
+            rowspans[col_idx] = rowspan
+            colspan: int = int(cell.attrs["colspan"])
+
+            # next column is offset by the colspan
+            span_offset += colspan - 1
+            value: str = ""
+            for item in cell.contents:
+                value += navigate_contents(item)
+
+            # clean the cell
+            value = value.strip().replace("\u2009", " ").replace("&#x000a0;", " ")
+            value = re.sub(r"\s", " ", value)
+            value = re.sub("<\\/?span[^>\n]*>?|<hr\\/>?", "", value)
+            value = re.sub("\\n", "", value)
+            if value.startswith("(") and value.endswith(")"):
+                value = value[1:-1]
+            if re.match(pval_regex, value):
+                value = re.sub(
+                    r"(\s{0,1})[*××xX](\s{0,1})10(_{0,1})", "e", value
+                ).replace("−", "-")
+            if re.match(pval_scientific_regex, value):
+                value = re.sub(r"(\s{0,1})[–−-](\s{0,1})", "-", value)
+                value = re.sub(r"(\s{0,1})[eE]", "e", value)
+            for drow, dcol in product(range(rowspan), range(colspan)):
+                try:
+                    table[row_idx + drow][col_idx + dcol] = value
+                    rowspans[col_idx + dcol] = rowspan
+                except IndexError:
+                    # rowspan or colspan outside the confines of the table
+                    pass
+
+        # update rowspan bookkeeping
+        rowspans = {c: s - 1 for c, s in rowspans.items() if s > 1}
+    return table
+
+
+def __check_superrow(row: list[str]) -> bool:
+    """Check if the current row is a superrow.
+
+    Args:
+        row (list): list of cells
+
+    Return:
+        True/False
+
+    """
+    cleaned_row: set[str] = set(
+        i for i in row if (str(i) != "") & (str(i) != "\n") & (str(i) != "None")
+    )
+    return len(cleaned_row) == 1 and bool(re.match("[a-zA-Z]", next(iter(cleaned_row))))
```

Comment:
How about we do a little spring cleaning while we're at it:

```suggestion
def __check_superrow(cells: list[str]) -> bool:
    """Check if the current row is a superrow.

    Args:
        cells: Cells in row
    """
    cleaned_row: set[str] = set(
        cell for cell in cells if cell not in ("", "\n", "None")
    )
    return len(cleaned_row) == 1 and bool(re.match("[a-zA-Z]", next(iter(cleaned_row))))
```

Do you know what a superrow is btw? I feel like this info should be in the docstring. (I know this isn't new code but was just wondering)

Reply:
Adding the superrow info to the doc string. A superrow is effectively where a row in the table will span further, such as having a row which contains cells split in some columns containing more data.